v1/src/main/java/com/google/cloud/teleport/spanner/common/SizedType.java

```diff
@@ -140,9 +171,16 @@ public static SizedType parseSpannerType(String spannerType, Dialect dialect) {
           }
           if (spannerType.startsWith("ARRAY<")) {
             // Substring "ARRAY<xxx>"
+            // Handle vector_length annotation
+            int annotationStart = spannerType.indexOf("(vector_length");
```

Comment:
Can you use regex to handle this parsing? It's not clear to a reader what the significance of this index is, or the `+16` below.